### DISCUSSION TITLE
html2canvas seems to re-download all photos in page despite receiving an element with not photos

### DISCUSSION THREAD

**Discussion by gnesherAugury** [niklasvh/html2canvas#3007] [Q&A]
Hi,
We are trying to take a screenshot of a graph and its surrounding elements (some text).

However, we can see in the network tab that when we execute the screenshot all images in the page are downloaded again even though they are not present in the screenshot.

What's more, when checking the request initiator (using Chrome) we can see that its html2canvas (and specifically:
documentClone.replaceChild(documentClone.adoptNode(this.documentElement), documentClone.documentElement);
)

A brief read through the code indicates that DocumentCloner intentionally clones the entire document, leading to this problem, but I'm not sure if this is intentional and if there's a way around it.

Our page has a number of large images and this slows the process of taking a screenshot

**Comment by wudizaliangbing** [comment]
Have you solved this problem? I have encountered the same problem

**Reply by zlac** [reply]
I solved with ignoreElements option

html2canvas(domElement, {
ignoreElements: function (element) {
						if (element.contains(domElement) 
						|| domElement.contains(element) 
						|| domElement === element 
						|| element.nodeName == 'HEAD' 
						|| element.nodeName == 'STYLE' 
						|| element.nodeName == 'META' 
						|| element.nodeName == 'LINK'
						) {
							return false;
						}else {
							return true;
						}
}})

**Reply by wudizaliangbing** [reply]
haha，thanks

**Reply by Angad-Godara** [reply]
Thanks @zlac, saved my day!

**Reply by leibaio** [reply]
@zlac Useful, help me a lot, tks 🙏

**Reply by mahipal123456** [reply]
thanks this alone make my capture speed too fast from 4 s to 400ms insanely fast thanks

**Comment by little077** [comment]
what fuck ？

**Reply by pancato** [reply]
?
